streaming/aggregate/sum.go

```diff
@@ -28,7 +32,11 @@ func (agg *Sum) AddValue(ctx context.Context, tx storage.StateTransaction, value
 	case octosql.TypeFloat:
 		currentSum = octosql.MakeFloat(currentSum.AsFloat() + value.AsFloat())
 	case octosql.TypeDuration:
-		currentSum = octosql.MakeDuration(currentSum.AsDuration() + value.AsDuration())
+		if currentSum.GetType() == octosql.TypeInt { // No value set
```

Comment:
This looks kinda terrible. Also, what if the previous values _were_ ints?

Reply:
Different records can have different types in a field. We have to check for that and throw an error accordingly.

Reply:
fixed type checking for empty valueState case

Reply:
added type checking, now it returns error if current value type and sum value type doesn't match